### DISCUSSION TITLE
vue-query useMutation() throws on error regardless of the throwOnError/useErrorBoundary value

### DISCUSSION THREAD

**Discussion by douglasg14b** [TanStack/query#7206] [Q&A]
I can't seem to figure out how to get vue-query to not throw on errors, the docs indicate that using `throwOnError: false` in the options should prevent this, but the behavior does not change as this value changes. (The same for `useErrorBoundary`)

In extra confusion, the typings for the mutation options shows `useErrorBoundary` and not `throwOnError`, however the v5 docs indicate that `useErrorBoundary` has been deprecated.

I'm on `1.26.0` and `vue 3.4.21`

```ts

useMutation({
        mutationFn: async (variables: TVariables) => {
            // If this throws, the exception propagates all the way to the UI
            return await doApiStuff(endpoint, variables)
        },
        throwOnError: false,
        useErrorBoundary: false,
        onSuccess,
        onError, // This is successfully called on error
    });
```

**Comment by Mini-ghost** [comment]
Could you please clarify the behavior you are expecting? Alternatively, could you provide an example of the minimal output you're looking for?

I wanted to point out that if an error occurs within `mutationFn`, an error will be thrown by the following code, and the `onError` callback will not be executed.

```ts
const { mutate } = useMutation({
  mutationFn() {
    return new Promise((_, reject) => {
      setTimeout(() => reject('error'), 300);
    });
  },
  onError() {
    console.log('error');
  },
  throwOnError: true,
});
```

I hope this helps clarify the issue.

**Reply by douglasg14b** [reply]
Hi, I'm expecting that `onError` will be called (which is is) and that the call to `mutationFn` will not receive an unhandled exception. Currently I need to wrap calls with a try/catch block to capture exceptions from calls to `mutationFn`.

**Reply by florjanklancar-kaldi** [reply]
I'm having the same problem has this been resolved?

**Reply by fvandenbulcke** [reply]
same here. Calling mutateAsync, onError is called but an error is throwed too

```javascript
const { mutateAsync } = useMutation({
  mutationFn() {
    throw new Error("Not implemented yet");
  },
  onError() {
    console.log('error');
  },
  throwOnError: false,
});
```
